include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -43,6 +45,40 @@ def TTNN_ToMemoryConfigOp : TTNN_Op<"to_memory_config"> {
     let hasVerifier = 1;
 }
 
+def TTNN_ToLayoutOp : TTNN_Op<"to_layout"> {
+    let summary = "ToLayout op.";
+    let description = [{
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         TTNN_LayoutAttr:$layout,
+                         TT_Device:$device);
+    let results = (outs AnyRankedTensor:$result);
+}
+
+def TTNN_ToDeviceOp : TTNN_Op<"to_device"> {
+    let summary = "ToDevice op.";
+    let description = [{
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         TT_Device:$device,
+                         TTNN_MemoryConfig:$memory_config);
+    let results = (outs AnyRankedTensor:$result);
+}
+
+// Constructor to MemoryConfig
+//
+def TTNN_MemoryConfigOp : TTNN_Op<"MemoryConfig"> {
```

Comment:
It _feels_ a bit strange to have a constructor defined as an op... @nsmithtt do you know of another way to do this?